safehttp/response_writer.go

```diff
@@ -53,20 +53,67 @@ func NotWritten() Result {
 	return Result{}
 }
 
-// Write TODO
+// Write dispatches the response to the Dispatcher, setting the Content-Type to
+// text/html; charset=utf-8 and response status to 200 OK if resp is a safe
+// response. The Dispatcher will then write the response to the underlying
+// Response Writer.
+//
+// TODO: replace panics with proper error handling when getting the response
+// Content-Type or writing the response fails.
 func (w *ResponseWriter) Write(resp Response) Result {
 	w.markWritten()
+	ct, err := w.d.ContentType(resp)
+	if err != nil {
+		panic(err)
+	}
+	w.rw.Header().Set("Content-Type", ct)
+	w.rw.WriteHeader(int(StatusOK))
 	if err := w.d.Write(w.rw, resp); err != nil {
-		panic("error")
+		panic(err)
 	}
 	return Result{}
 }
 
-// WriteTemplate TODO
+// WriteJSON encapsulates data into a JSON response and dispatches it to the
+// Dispatcher, to be serialised and written to the ResponseWriter. It will also
+// set the Content-Type to application/json; charset=utf-8 and the response
+// status to 200 OK.
+//
+// TODO: replace panics with proper error handling when getting the response
+// Content-Type or writing the response fails.
+func (w *ResponseWriter) WriteJSON(data interface{}) Result {
+	w.markWritten()
+	resp := JSONResponse{Data: data}
+	ct, err := w.d.ContentType(resp)
+	if err != nil {
+		panic(err)
+	}
+	w.rw.Header().Set("Content-Type", ct)
+	w.rw.WriteHeader(int(StatusOK))
+	if err := w.d.WriteJSON(w.rw, resp); err != nil {
+		panic(err)
+	}
+	return Result{}
+}
+
+// WriteTemplate dispatches a parsed template and a data object to the
+// Dispatcher to be executed and written to the underlying Response Writer, in
+// case the template is a safe HTML template. If t is a safe HTML Template, the
+// Content-Type will also be set to text/html; charset=utf-8 and the response
+// status to 200 OK.
+//
+// TODO: replace panics with proper error handling when getting the response
+// Content-Type or writing the response fails.
```

Comment:
The `Content-Type` for safe html templates depends on the Dispatcher implementation and should be left out of this comment.